File: src/tools/logging.rs
use std::fmt::*;

// def info(s):
//     print("=( INFO )====>", s)

#[allow(dead_code)]
pub fn info<T: Display>(s: T) {
    println!("=( INFO )====> {}", s);
}

// def debug(s):
//     print("=| DEBUG |===>", s)
#[allow(dead_code)]
pub fn debug<T: Display>(s: T) {
    println!("=| DEBUG |===> {}", s);
}

// def warning(s):
//     print("={ WARNING }=>", s)
#[allow(dead_code)]
pub fn warning<T: Display>(s: T) {
    println!("={{ WARNING }}=> {}", s);
}

// def error(s):
//     print("=[ ERROR ]===>", s)
#[allow(dead_code)]
pub fn error<T: Display>(s: T) {
    println!("=[ ERROR ]===> {}", s);
}

File: examples/importing/logging.m
Log = Log_A.Log_B.(
    PipeFn[c.(PipeFn[a.(PipeStrln[Log_B])][PipeStr[Log_A]])][Log_B]
)

Info = Info_A.(Log["=( INFO )====> "][Info_A])
Debug = Debug_A.(Log["=| DEBUG |===> "][Debug_A])
Warning = Warning_A.(Log["={ WARNING }=> "][Warning_A])
Error = Error_A.(Log["=[ ERROR ]===> "][Error_A])

Good = True
Bad  = Bad_A.Bad_B.Bad_B